server/routes/oauth2s.js

```diff
@@ -1,54 +1,36 @@
-const bcrypt = require('bcrypt');
-const { nanoid } = require('nanoid/async');
-const fetch = require('node-fetch');
-const Router = require('koa-router');
 
+const Router = require('koa-router');
 const User = require('../models/user');
 const Oauth2 = require('../models/oauth2');
+const authStrategies = require('../utils/authStrategies/index');
+const log = require('../utils/logger');
 const GroupMembers = require('../models/group_members');
 
 const router = new Router({
   prefix: '/oauth2s'
 });
 
 router.post('/', async ctx => {
-  const googleToken = ctx.request.body.oauth2.googleToken;
-  const provider = ctx.request.body.oauth2.provider;
-  const response = await fetch(`https://people.googleapis.com/v1/people/me?access_token=${googleToken}&personFields=names,emailAddresses`);
-  const data = await response.text();
-  const gData = JSON.parse(data);
-
-  const hashPassword = await bcrypt.hash(googleToken, 10);
-  const username = await nanoid(11);
-  // const username = gData.names[0].displayName.split(' ').join('').replace(/^[\s\uFEFF\xAO] + |[\s\uFEFF\xAO] + $ /g, ' ');
-  let newUser = {
-    email: gData.emailAddresses[0].value,
-    hash: hashPassword,
-    username: username.toLowerCase(),
-    firstName: gData.names[0].familyName,
-    lastName: gData.names[0].givenName,
-    emailVerified: true,
-    resetPasswordExpires: new Date(),
-    metadata: {
-      'profileComplete': 'false',
-      'oneInviteComplete': 'false',
-      'firstName': gData.names[0].familyName,
-      'lastName': gData.names[0].givenName
-    }
-  };
-
+  const { code, provider } = ctx.request.body.oauth2;
+  const newUser = await authStrategies[provider](code);
   try {
     const user = await User.query().insertAndFetch(newUser);
     await Oauth2.query().insert({ provider: provider, email: newUser.email, user_id: user.id });
-    await GroupMembers.query().insert({ user_id: user.id, group_id: 'groupVerified' });
+    await GroupMembers.query().insert({ user_id: user.id, group_id: 'groupBasic' });
```

Comment:
this will fail because insert will return an identifier where in this case will default to `id` column value that does not exist